<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Metabolomics – MS-DIAL 5 tutorial</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-e26003cea8cd680ca0c55a263523d882.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-659650fc26dc25888fc1474f317bb8ac.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">MS-DIAL 5 tutorial</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./lipidomics.html"> 
<span class="menu-text">Lipidomics</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./metabolomics.html" aria-current="page"> 
<span class="menu-text">Metabolomics</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./proteomics.html"> 
<span class="menu-text">Proteomics</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./imaging.html"> 
<span class="menu-text">Mass spectrometry imaging</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-utilities" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Utilities</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-utilities">    
        <li>
    <a class="dropdown-item" href="./utils.html">
 <span class="dropdown-text">Other utilities</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./duplicate_alignment_spotfeature.html">
 <span class="dropdown-text">Duplicate alignment spot feature to separate isomers</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./Determining MS data types.html">
 <span class="dropdown-text">Determining Mass Spectrometry (MS) Data Types in MS-Dial: Centroid or Profile</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./Determine the optimal minimum peak height.html">
 <span class="dropdown-text">Determining optimal minimum peak height</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./Explanation on each parameter of MS-Dial 5.html">
 <span class="dropdown-text">Tutorial on MS-Dial 5</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="./about.html"> 
<span class="menu-text">About</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#gc-ms-nominal-mass" id="toc-gc-ms-nominal-mass" class="nav-link active" data-scroll-target="#gc-ms-nominal-mass">1. GC-MS (nominal mass)</a>
  <ul class="collapse">
  <li><a href="#imput-data" id="toc-imput-data" class="nav-link" data-scroll-target="#imput-data">1-1 Imput data</a></li>
  <li><a href="#starting-up-your-project" id="toc-starting-up-your-project" class="nav-link" data-scroll-target="#starting-up-your-project">1-2 Starting up your project</a></li>
  <li><a href="#setting-parameters" id="toc-setting-parameters" class="nav-link" data-scroll-target="#setting-parameters">1-3 Setting parameters</a></li>
  <li><a href="#result-of-annotation" id="toc-result-of-annotation" class="nav-link" data-scroll-target="#result-of-annotation">1-4 Result of annotation</a></li>
  </ul></li>
  <li><a href="#gc-ms-high-resolution" id="toc-gc-ms-high-resolution" class="nav-link" data-scroll-target="#gc-ms-high-resolution">2. GC-MS (high resolution)</a>
  <ul class="collapse">
  <li><a href="#imput-data-1" id="toc-imput-data-1" class="nav-link" data-scroll-target="#imput-data-1">2-1 Imput data</a></li>
  <li><a href="#setting-parameters-1" id="toc-setting-parameters-1" class="nav-link" data-scroll-target="#setting-parameters-1">2-2 Setting parameters</a></li>
  <li><a href="#result-of-anotation" id="toc-result-of-anotation" class="nav-link" data-scroll-target="#result-of-anotation">2-3 Result of anotation</a></li>
  </ul></li>
  <li><a href="#lcms-or-lcmsms-data-dependent-msms-project-with-user-defined-msms-database-msp-format" id="toc-lcms-or-lcmsms-data-dependent-msms-project-with-user-defined-msms-database-msp-format" class="nav-link" data-scroll-target="#lcms-or-lcmsms-data-dependent-msms-project-with-user-defined-msms-database-msp-format">3. LC/MS or LC/MS/MS (data dependent MS/MS) project with user-defined MS/MS database (MSP format)</a>
  <ul class="collapse">
  <li><a href="#input-data" id="toc-input-data" class="nav-link" data-scroll-target="#input-data">Input Data</a></li>
  <li><a href="#the-video-tutorial-for-the-ms-dial-operation" id="toc-the-video-tutorial-for-the-ms-dial-operation" class="nav-link" data-scroll-target="#the-video-tutorial-for-the-ms-dial-operation">The video tutorial for the MS-DIAL operation</a></li>
  <li><a href="#key-points-in-the-ms-dial-operation" id="toc-key-points-in-the-ms-dial-operation" class="nav-link" data-scroll-target="#key-points-in-the-ms-dial-operation">Key points in the MS-DIAL operation</a></li>
  </ul></li>
  <li><a href="#ce-msms" id="toc-ce-msms" class="nav-link" data-scroll-target="#ce-msms">3. CE-MS/MS</a>
  <ul class="collapse">
  <li><a href="#the-video-tutorial-for-the-ms-dial-operation-1" id="toc-the-video-tutorial-for-the-ms-dial-operation-1" class="nav-link" data-scroll-target="#the-video-tutorial-for-the-ms-dial-operation-1">The video tutorial for the MS-DIAL operation</a></li>
  </ul></li>
  <li><a href="#ms-dial-isotope-tracking-function" id="toc-ms-dial-isotope-tracking-function" class="nav-link" data-scroll-target="#ms-dial-isotope-tracking-function">4. MS-DIAL isotope tracking function</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Metabolomics</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>See <a href="./index.html#fig-2">Figure 2</a> for the meanings of the abbreviations used in this chapter.</p>
<section id="gc-ms-nominal-mass" class="level2">
<h2 class="anchored" data-anchor-id="gc-ms-nominal-mass">1. GC-MS (nominal mass)</h2>
<section id="imput-data" class="level3">
<h3 class="anchored" data-anchor-id="imput-data">1-1 Imput data</h3>
<p>This tutorial uses a GC-MS dataset from the Development of Metabolite Profiling Database for Knock-Out Mutants in Arabidopsis. The dataset for reproducing this tutorial can be obtained from <a href="https://doi.org/10.5281/zenodo.10604540"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.10604540.svg" class="img-fluid" alt="DOI"></a></p>
</section>
<section id="starting-up-your-project" class="level3">
<h3 class="anchored" data-anchor-id="starting-up-your-project">1-2 Starting up your project</h3>
<p><img src="images/gcnominal/1-1.png" class="img-fluid" alt="fig1-1"> <img src="images/gcnominal/1-2.png" class="img-fluid" alt="fig1-2"> <img src="images/gcnominal/1-3.png" class="img-fluid" alt="fig1-3"> <img src="images/gcnominal/1-4.png" class="img-fluid" alt="fig1-4"></p>
<p>(1)Launch MS-DIAL,and select New project.<br>
(2)Enter the project title.Then,enter of the path of the folder containing the data you want to analyze in the project path. Click Next.<br>
(3)Select the file format you want to input and input file. Click Next.</p>
</section>
<section id="setting-parameters" class="level3">
<h3 class="anchored" data-anchor-id="setting-parameters">1-3 Setting parameters</h3>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/gcnominal/2-1.png" class="img-fluid figure-img"></p>
<figcaption>fig2-1</figcaption>
</figure>
</div>
<p>(1)After confirming that you have imported the data, click Next.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/gcnominal/2-2ver2.png" class="img-fluid figure-img"></p>
<figcaption>fig2-2</figcaption>
</figure>
</div>
<p>(2)Next, set the measurement parameters. For GC-MS, select <strong>Hard ionization</strong> type item. Select the ion mode that applies to your experiment.</p>
<p>The following table shows the parameters changed from the defaults in this analysis.　　　　</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">parameter</th>
<th style="text-align: left;">example</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">measurement parameters</td>
<td style="text-align: left;">Ionization type</td>
<td style="text-align: left;">Hard ionization</td>
</tr>
<tr class="even">
<td style="text-align: left;">measurement parameters</td>
<td style="text-align: left;">Ion mode</td>
<td style="text-align: left;">Positive ion mode</td>
</tr>
<tr class="odd">
<td style="text-align: left;">measurement parameters</td>
<td style="text-align: left;">target omics</td>
<td style="text-align: left;">Metabolomics</td>
</tr>
</tbody>
</table>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/gcnominal/2-3.png" class="img-fluid figure-img"></p>
<figcaption>fig2-3</figcaption>
</figure>
</div>
<p>(3)In the peak detection section, you can change the parameters for peak detection. The data used in this tutorial are nominal mass data, so Accurate MS is <strong>not</strong> checked.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/gcnominal/2-4.png" class="img-fluid figure-img"></p>
<figcaption>fig2-4</figcaption>
</figure>
</div>
<p>(4)Configure identification setting.Please choose whether to use retention index or retention time for annotation. If you use retention index, click Set and designate the PATH of the retention index to be used.If there are library you want to reference, click Browse and designate the library.　　</p>
</section>
<section id="result-of-annotation" class="level3">
<h3 class="anchored" data-anchor-id="result-of-annotation">1-4 Result of annotation</h3>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/gcnominal/fig3.png" class="img-fluid figure-img"></p>
<figcaption>fig3</figcaption>
</figure>
</div>
<p>This is the main viewer of MS-DIAL. (1) Double-clicking on a file name in the file navigator will display the detected peak information in the center window(Peak spot viewer).Blue spots indicate peaks of lower abundance in the sample, red spots indicate peaks of high abundance, and green spots indicate peaks of intermediate abundance. (2) Click on the triangle at the top of the Peak spot viewer to display the compound corresponding to the retention time.(3) The MS1 spectrum of the focused peak is displayed in the left window of the Peak spot viewer. (4)The ion chromatogram of the focused peak is displayed in the upper window.(5)Other peak information is displayed in the upper right window.</p>
</section>
</section>
<section id="gc-ms-high-resolution" class="level2">
<h2 class="anchored" data-anchor-id="gc-ms-high-resolution">2. GC-MS (high resolution)</h2>
<section id="imput-data-1" class="level3">
<h3 class="anchored" data-anchor-id="imput-data-1">2-1 Imput data</h3>
<p><img src="images/gcnominal/fig4.png" class="img-fluid" alt="fig4"> The data used in this tutorial are the results of a non-targeted GC-HRMS (Q Exactive GC Orbitrap) analysis of a PM2.5 air sample from the Maldives. This data is high-resolution mass spectrometry data.The files used can be downloaded from <a href="https://massive.ucsd.edu/ProteoSAFe/dataset_files.jsp?task=d98091ce8b2c4e4dba7199934341ef57#%7B%22table_sort_history%22%3A%22main.collection_asc%22%7D">MassIVE MSV000087683 Files</a>.</p>
</section>
<section id="setting-parameters-1" class="level3">
<h3 class="anchored" data-anchor-id="setting-parameters-1">2-2 Setting parameters</h3>
<p><img src="images/gcnominal/fig5.png" class="img-fluid" alt="fig5"> (1)Launch MS-DIAL,and select New project.<br>
(2)Enter a project name. Enter the folder containing the data you want to input into the project file path. Then click Next.<br>
(3)Next,select the analysis file path. First click on Browse.<br>
(4)Open the folder containing the files you wish to analyze, select the files and click Open. The same operation can be done with drag-and-drop. Also, remember to select the file format in the lower right corner according to the format of the file you wish to enter.<br>
<img src="images/gcnominal/6-5.png" class="img-fluid" alt="fig6-1"> <img src="images/gcnominal/6-6.png" class="img-fluid" alt="fig6-2"> <img src="images/gcnominal/6-7.png" class="img-fluid" alt="fig6-3"> <img src="images/gcnominal/6-8.png" class="img-fluid" alt="fig6-4"> (5)After inputting the file, you will see a screen like this. When you confirm that the file has been entered, press the NEXT button to proceed.。<br>
(6)Here we set the measurement parameter.In this case, select <strong>Hard ionization</strong> for GCMS. Select data type, ion mode, and targetomics according to the data you want to analyze, and click Next.<br>
(7)The data collection section enable you to determine the mass range and retention time range you want to observe.<br>
(8)In the peak detection section, you can change the parameters for peak detection. The data used in this tutorial are high resolution data, so <strong>Accurate MS is checked</strong>.<br>
<img src="images/gcnominal/7-9.png" class="img-fluid" alt="fig7-1"> <img src="images/gcnominal/7-10.png" class="img-fluid" alt="fig7-2"> <img src="images/gcnominal/7-11.png" class="img-fluid" alt="fig7-3"> (9)The Spectrum deconvolution section allows you to set parameters for deconvolution of the spectrum.<br>
(10)Please choose whether to use <strong>retention index</strong> or <strong>retention time</strong> for annotation. If you use retention index, you need to set up index file.This tutorial uses retention time.<br>
(11)Finally, set the alignment parameters and click Run.　　</p>
<div class="callout callout-style-simple callout-note no-icon">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-body-container">
<p>If you use a retention index,make sure you have the right retention index for the data you want to analyze.</p>
</div>
</div>
</div>
<p>The following table shows the parameters changed from the defaults in this analysis.　　</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">parameter</th>
<th style="text-align: left;">example</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">measurement parameters</td>
<td style="text-align: left;">Ionization type</td>
<td style="text-align: left;">Hard ionization</td>
</tr>
<tr class="even">
<td style="text-align: left;">measurement parameters</td>
<td style="text-align: left;">Ion mode</td>
<td style="text-align: left;">Positive ion mode</td>
</tr>
<tr class="odd">
<td style="text-align: left;">measurement parameters</td>
<td style="text-align: left;">target omics</td>
<td style="text-align: left;">Metabolomics</td>
</tr>
<tr class="even">
<td style="text-align: left;">Peak detection</td>
<td style="text-align: left;">Accurate MS</td>
<td style="text-align: left;">Check</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Identification</td>
<td style="text-align: left;">RI or RT</td>
<td style="text-align: left;">RI</td>
</tr>
</tbody>
</table>
</section>
<section id="result-of-anotation" class="level3">
<h3 class="anchored" data-anchor-id="result-of-anotation">2-3 Result of anotation</h3>
<p><img src="images/gcnominal/fig8.png" class="img-fluid" alt="fig8"><br>
This is the main viewer of MS-DIAL.(1) Double-clicking on a file name in the file navigator will display the detected peak information in the center window.Then select the Alignment spot viewer next to the Peak spot viewer to see the alignment results.Blue spots indicate peaks of lower abundance in the sample, red spots indicate peaks of high abundance, and green spots indicate peaks of intermediate abundance. (2)Click on the spot in the center window to see the information corresponding to that peak spot. (3) The MS1 spectrum of the focused peak is displayed in the left window of the center window.(4)The ion chromatogram of the focused peak is displayed in the upper window. (5) Other peak information is displayed in the upper right window.　　</p>
</section>
</section>
<section id="lcms-or-lcmsms-data-dependent-msms-project-with-user-defined-msms-database-msp-format" class="level2">
<h2 class="anchored" data-anchor-id="lcms-or-lcmsms-data-dependent-msms-project-with-user-defined-msms-database-msp-format">3. LC/MS or LC/MS/MS (data dependent MS/MS) project with user-defined MS/MS database (MSP format)</h2>
<p>Here, a project involving data-dependent MS/MS acquisition in conjunction with a user-defined MSP library (a composite library comprising MassBank, GNPS, and Respect) is presented.</p>
<section id="input-data" class="level3">
<h3 class="anchored" data-anchor-id="input-data">Input Data</h3>
<p>This section utilizes a total of 6 ABF files, with the MSP file being located in the same ZIP folder as this demonstration. The ABF files for this demonstration can be downloaded from <a href="http://prime.psc.riken.jp/compms/msdial/download/demo/20160805-Wine-DDA-Pos-Demo%20files.zip" class="uri">http://prime.psc.riken.jp/compms/msdial/download/demo/20160805-Wine-DDA-Pos-Demo%20files.zip</a>. The experimental protocol has been previously described in the following research: <a href="http://pubs.acs.org/doi/abs/10.1021/acs.jafc.5b04890" class="uri">http://pubs.acs.org/doi/abs/10.1021/acs.jafc.5b04890</a>.</p>
<section id="experiment-summary" class="level4">
<h4 class="anchored" data-anchor-id="experiment-summary">Experiment summary:</h4>
<ul>
<li><strong>Liquid chromatography:</strong> total 4 min run per sample with Kinetex C18 2.6 μm (50×1.0 mm)</li>
<li><strong>Solvent A:</strong> water with 0.1% acetic acid</li>
<li><strong>Solvent B:</strong> acetonitrile with 0.1% acetic acid</li>
<li><strong>Mass spectrometer:</strong> data dependent method with positive ion mode
<ul>
<li>Collision energy: 35 V</li>
<li>Collision energy spread: 15 V</li>
<li>Cycle time: 125 ms</li>
<li>Mass range: m/z 60-1250</li>
</ul></li>
</ul>
</section>
</section>
<section id="the-video-tutorial-for-the-ms-dial-operation" class="level3">
<h3 class="anchored" data-anchor-id="the-video-tutorial-for-the-ms-dial-operation">The video tutorial for the MS-DIAL operation</h3>
<div class="quarto-video ratio ratio-16x9"><iframe data-external="1" src="https://www.youtube.com/embed/cZY7wqF5Wdo" title="" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe></div>
</section>
<section id="key-points-in-the-ms-dial-operation" class="level3">
<h3 class="anchored" data-anchor-id="key-points-in-the-ms-dial-operation">Key points in the MS-DIAL operation</h3>
<ol type="1">
<li>All sample <code>Type</code>s are <code>Sample</code>. The input data here does not contain <code>QC</code> or <code>Blank</code>.</li>
<li>For this input data, set a different <code>Class ID</code> for each sample.</li>
<li>You can paste Excel cells to MS-DIAL sample table.</li>
</ol>
</section>
</section>
<section id="ce-msms" class="level2">
<h2 class="anchored" data-anchor-id="ce-msms">3. CE-MS/MS</h2>
<p>A project dealing with CE-MS/MS data measured using vDIA(variable data independent acquisition) in combination with migration time correction is demonstrated.</p>
<p>This tutorial uses 19 demonstration data files and three libralies which are downloadable from the below link.<br>
</p>
<p><a href="http://prime.psc.riken.jp/compms/msdial/download/demo/dims.zip">https://doi.org/10.5281/zenodo.10597056/dims.zip</a></p>
<section id="experiment-summary-1" class="level4">
<h4 class="anchored" data-anchor-id="experiment-summary-1"><strong>Experiment summary:</strong><br>
</h4>
<ul>
<li>Hydrophilic metabolites were extracted from RAW264.7 cells according to previously described protocol. (Human metabolome Technologies Inc.&nbsp;Patent JP WO2014045932 A1)</li>
<li>Capillary electrophoresis was performed using a CE System 7100(Agilent Technologies Inc.), and an Agilent 1260 isocratic HPLC pump was used to create the sheath flow.</li>
<li>The MS/MS spectrum was acquired using Q Exactive Plus (Thermo Fisher Scientific Inc.) with vDIA</li>
</ul>
</section>
<section id="note" class="level4">
<h4 class="anchored" data-anchor-id="note"><strong>Note:</strong><br>
</h4>
<ul>
<li><strong>Txt format library</strong></li>
<li>In this tutorial, two txt files, “Identification_anion.txt” and “MTcorrection_anion.txt”, were used for identification of internal standards and migration time correction, respectively.</li>
<li>See below link for the details of library format: <a href="http://prime.psc.riken.jp/compms/msdial/main.html#Templates" class="uri">http://prime.psc.riken.jp/compms/msdial/main.html#Templates</a></li>
<li><strong>Migration time (MT) correction</strong></li>
<li>Migration times of metabolite ion peaks were linearly corrected along the MT axis using internal standard compounds.</li>
<li>If migration times of the reference compounds in each file doesn’t match reference MT, change the parametors from reference compound informatin table and try MT correction again.</li>
</ul>
</section>
<section id="the-video-tutorial-for-the-ms-dial-operation-1" class="level3">
<h3 class="anchored" data-anchor-id="the-video-tutorial-for-the-ms-dial-operation-1">The video tutorial for the MS-DIAL operation</h3>
<div class="quarto-video ratio ratio-16x9"><iframe data-external="1" src="https://www.youtube.com/embed/T9krZs2jNSs" title="" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe></div>
</section>
</section>
<section id="ms-dial-isotope-tracking-function" class="level2">
<h2 class="anchored" data-anchor-id="ms-dial-isotope-tracking-function">4. MS-DIAL isotope tracking function</h2>
<p>This feature is currently in the development stage in MS-DIAL5, so if you want to use it now, please use MS-DIAL4.</p>
<p><a href="https://systemsomicslab.github.io/mtbinfo.github.io/MS-DIAL/tutorial#chapter-7">https://systemsomicslab.github.io/mtbinfo.github.io/MS-DIAL/tutorial#chapter-7</a></p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/systemsomicslab\.github\.io\/msdial5tutorial\/");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>